src/bundle/Controller/LocationController.php

```diff
@@ -83,6 +85,26 @@ public function actionsAction(
         return $this->reloadTab('locations', $content->id, $redirectLocationId);
     }
 
+    public function swapLocationAction(
+        Content $content,
+        Location $location,
+        Request $request
+    ) {
+        $swapLocationsForm = $this->formFactory->createLocationsContentSwapForm();
+        $swapLocationsForm->handleRequest($request);
+
+        if ($swapLocationsForm->isValid()) {
+            $newLocationId = $swapLocationsForm->get('new_location_id')->getData();
+            $this->uiLocationService->swapLocations($location, $newLocationId);
+
+            return $this->resetLocation($content->id);
+        }
+
+        $redirectLocationId = $request->query->get('redirectLocationId', $content->contentInfo->mainLocationId);
+
+        return $this->reloadTab('locations', $content->id, $redirectLocationId);
```

Comment:
@SylvainGuittard has replied on JIRA. Please change this to a `resetLocation()` with the source location (`$location`).